spec/datadog/appsec/security_engine/runner_spec.rb

```diff
@@ -171,4 +178,19 @@
       end
     end
   end
+
+  describe '#finalize!' do
+    it 'releases the waf handle when an error occurs' do
+      # waf_context is lazily initialized, so we need to ensure it is called
+      runner.waf_context
+
+      expect(waf_context).to receive(:finalize!).and_raise(StandardError)
+      expect(thread_safe_ref).to receive(:release).with(waf_handle)
+
+      begin
+        runner.finalize!
+      rescue
+      end
```

Comment:
This one is better to be normal expectation of exception

Reply:
agree, fixed [4475813](https://github.com/DataDog/dd-trace-rb/pull/4833/commits/44758138273d01f5d743d74b2e071c15d197b9c8)